pkg/kvstore/kvstore.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kvstore
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"sync"
+
+	"github.com/vmware/vic/pkg/trace"
+)
+
+var (
+	ErrKeyNotFound = errors.New("key not found")
+)
+
+// This package implements a very basic key/value store.  It is up to the
+// caller to provision the namespace.
+
+type KeyValueStore struct {
+	b Backend
+
+	kv map[string][]byte
+
+	fileName string
+
+	l sync.Mutex
+}
+
+type Backend interface {
```

Comment:
Sure, if you had access to VFS.  We don't.  Remember, this goes to VC, then to the datastore, which may or may not be eventually consistent.  Where are you writing the data, how do you know if your last write was complete?